From: Johan Hovold <johan@kernel.org>
Date: Wed, 16 Nov 2016 15:20:37 +0100
Subject: of_mdio: fix device reference leak in of_phy_find_device
Patch-mainline: v4.9-rc7
Git-commit: 3ae30f4ce65e9d4de274b1472169ab3c27f5c666
References: bsc#1026030 FATE#321670

Make sure to drop the reference taken by bus_find_device() before
returning NULL from of_phy_find_device() when the found device is not a
PHY.

Fixes: 6ed742363b9c ("of: of_mdio: Ensure mdio device is a PHY")
Signed-off-by: Johan Hovold <johan@kernel.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Benjamin Poirier <bpoirier@suse.com>
---
 drivers/of/of_mdio.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/of/of_mdio.c
+++ b/drivers/of/of_mdio.c
@@ -294,6 +294,7 @@ struct phy_device *of_phy_find_device(st
 		mdiodev = to_mdio_device(d);
 		if (mdiodev->flags & MDIO_DEVICE_FLAG_PHY)
 			return to_phy_device(d);
+		put_device(d);
 	}
 
 	return NULL;
